### PR TITLE
Add examples

### PR DESCRIPTION
**Pull request by sb8244**
This adds support for the example property on swagger 1.2 models and model properties. They will map over the the identical field in swagger 2.0 definitions.

https://github.com/apigee-127/swagger-converter/issues/61

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -688,25 +688,38 @@ prototype.buildModel = function(oldModel) {
...
     }
 
+    if (oldProperty.hasOwnProperty('example')) {
+      allowedProperties.example = oldProperty.example;
+    }
+
     properties[propertyName] = extend({},
       this.buildDataType(oldProperty, true),
-      {description: oldProperty.description}
+      allowedProperties
```
Comment (IvanGoncharov) [inline index.js]: You don't need such big changes.
Just add `example: undefinedIfEmpty(oldProperty.example)` in two places
Here

File: index.js [hunk 2]
```diff
@@ -688,25 +688,38 @@ prototype.buildModel = function(oldModel) {
...
 
   required = oldModel.required || required;
 
-  return extend(this.buildDataType(oldModel, true),
-  {
+  var allowedProperties = {
     description: oldModel.description,
     required: undefinedIfEmpty(required),
     properties: undefinedIfEmpty(properties),
     discriminator: oldModel.discriminator
```
Comment (IvanGoncharov) [inline index.js]: and also here

File: index.js (hunk 1 above)
Comment (sb8244) [inline index.js]: Will that make example key be present even if it were undefined?

Comment (IvanGoncharov) [inline index.js]: No, only if it proper value.
For example it also skip null and empty string

Comment (IvanGoncharov) [inline index.js]: You can see check here https://github.com/apigee-127/swagger-converter/blob/master/index.js#L824

Comment (sb8244) [inline index.js]: That will put an undefined value for the example key, no?

On Thursday, November 19, 2015, Ivan Goncharov <notifications@github.com>
wrote:

> In index.js
> <https://github.com/apigee-127/swagger-converter/pull/62#discussion_r45416240>
> :
>
> >      properties[propertyName] = extend({},
> >        this.buildDataType(oldProperty, true),
> > -      {description: oldProperty.description}
> > +      allowedProperties
>
> You can see check here
> https://github.com/apigee-127/swagger-converter/blob/master/index.js#L824
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apigee-127/swagger-converter/pull/62/files#r45416240>.
>


-- 


  Steve Bussey •SalesLoft

•Software Engineer

e steve.bussey@salesloft.com w salesloft.com
<http://salesloft.com/?v=1&utm_expid=76406993-1.E5ncJhwyTAW9uGaR1WdBTA.1?utm_source=Email-Signature-Rescue&utm_medium=Email-Signature&utm_campaign=Email-Signature-Rescue>

a 3423 Piedmont Rd NE, Atlanta, GA 30305

[image: Facebook] <https://www.facebook.com/SalesLoft> [image: Twitter]
<https://twitter.com/SalesLoft> [image: Linkedin]
<https://www.linkedin.com/company/salesloft> [image: wordpress]
<http://blog.salesloft.com/blog/> [image:
http://salesloft.com/prospector/#video]
<http://salesloft.com/prospector/#video>  [image: Software Engineer Logo]
<http://salesloft.com/?v=1&utm_expid=76406993-1.E5ncJhwyTAW9uGaR1WdBTA.1?utm_source=Email-Signature-Rescue&utm_medium=Email-Signature&utm_campaign=Email-Signature-Rescue>

Comment (IvanGoncharov) [inline index.js]: @sb8244 Sorry I made mistake. You don't need `undefinedIfEmpty` wrapper.
simply `example: oldProperty.example`.

Comment (IvanGoncharov) [inline index.js]: `extend` function checks values internally, see: https://github.com/apigee-127/swagger-converter/blob/master/index.js#L798

Comment (IvanGoncharov) [inline index.js]: We convert a lot of optional fields so to check each of them individually mean huge code bloat.
At least 3 lines instead of single one.

**Comment by mohsen1** [comment]
@IvanGoncharov please review and let me know when it's good to merge.

File: test/output/complex-models.json [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+      "get": {
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/Project"
+            }
+          }
+        },
+        "parameters": [{
```
Comment (IvanGoncharov) [inline test/output/complex-models.json]: @sb8244 
You have custom formatting of ouput file so it require manual fixing each time we change something in the lib.
Please uncomment [this lines](https://github.com/apigee-127/swagger-converter/blob/master/test/test.js#L107-L108) run test and copy `complex-models.json-converted` file.

@mohsen1 What do you think about doing it similar to [this](https://github.com/lucybot/api-spec-converter/blob/master/test/convert.js#L61-L62)?

**Comment by IvanGoncharov** [comment]
@sb8244 Also can you also squash  your changes into one commit, it will simplify history.
Not mandatory, but would be nice if you split code formatting(removing empty line, ...) into separate commit.

**Comment by sb8244** [comment]
I would be happy to squash

**Comment by sb8244** [comment]
@IvanGoncharov I have squashed & fixed the json formatting

**Comment by IvanGoncharov** [comment]
@sb8244 Thanks
@mohsen1 :+1: ready for merge

**Comment by mohsen1** [comment]
Thank you! 😀

**Comment by sb8244** [comment]
Thank you both! I'll PR again if I find any weirdness. 

Flow right now is swagger-docs ruby bindings to swagger 1.2 json files, then running a conversion script to go from 1.2 to 2.0. From there, pipe it into a themed swagger-ui.

**Comment by IvanGoncharov** [comment]
@sb8244 SwaggerUI has native support for 1.2(it has its own converter).
Did you try that?
If so, what was your experience?

**Comment by sb8244** [comment]
I have been using https://github.com/jensoleg/swagger-ui which is a much more production ready documentation visual than the default theme. This theme *does* support 1.2, but the examples section isn't processed unless it's in 2.0, because of the change from model to definitions.

**Pull request merged**